src/status_im/utils/universal_links/core.cljs

```diff
@@ -81,17 +81,20 @@
   (log/info "universal-links: handling community" community-id)
   (navigation/navigate-to cofx :community {:community-id community-id}))
 
-
 (rf/defn handle-navigation-to-desktop-community-from-mobile
   {:events [:handle-navigation-to-desktop-community-from-mobile]}
   [cofx deserialized-key]
-  (navigation/navigate-to cofx :community-overview deserialized-key))
+  (rf/merge
+   cofx
+   {:dispatch [:navigate-to :community-overview deserialized-key]}
+   (navigation/pop-to-root :shell-stack)))
```

Comment:
Hi @Parveshdhull, I saw you deleted the `navigate-to-community` event here https://github.com/status-im/status-mobile/pull/17586/files#diff-f3a5b4391070354e91f7f3e37afe74dee7a9653180c0dc18a2f1fbd2c85ac1dcL786

and it looks like the below pattern should be avoided
```clojure
{:dispatch [:navigate-to :community-overview deserialized-key]}
(navigation/pop-to-root :shell-stack)
```

`handle-navigation-to-desktop-community-from-mobile` is used when open universal links in app or triggered by universal link outside app

Is this pattern the right approach for this situation?

Reply:
Hi @yqrashawn, yeah it looks good to me.
Also, I removed that event because it wasn't used anywhere, feel free to add it back.